uefi-raw/src/protocol/media.rs

```diff
@@ -2,9 +2,60 @@ use crate::protocol::device_path::DevicePathProtocol;
 use crate::{guid, Guid, Status};
 use core::ffi::c_void;
 
+/// Load File Protocol.
+///
+/// Used to obtain files, that are primarily boot options, from arbitrary
+/// devices.
+///
+/// # UEFI Spec Description
+/// The EFI_LOAD_FILE_PROTOCOL is a simple protocol used to obtain files from
+/// arbitrary devices.
+///
+/// When the firmware is attempting to load a file, it first attempts to use the
+/// device’s Simple File System protocol to read the file. If the file system
+/// protocol is found, the firmware implements the policy of interpreting the
+/// File Path value of the file being loaded. If the device does not support the
+/// file system protocol, the firmware then attempts to read the file via the
+/// EFI_LOAD_FILE_PROTOCOL and the LoadFile() function. In this case the
+/// LoadFile() function implements the policy of interpreting the File Path
+/// value.
```

Comment:
nit: While I don't mind having documentation here, we haven't generally bothered with docstrings in `uefi-raw` except for cases where the type is publically re-exported from `uefi`.

Reply:
got it. Perhaps we can add links to the UEFI web/html documentation eventually.